apps/extension/src/wasm-build-action.ts

```diff
@@ -38,6 +39,10 @@ const workerListener = ({ data }: { data: ActionBuildRequest }) => {
 
 self.addEventListener('message', workerListener, { once: true });
 
+const actionTypeRequiresProvingKey = (
```

Comment:
My logic for whether a proving key is required for a given action, is checking whether such a proving key even exists. Let me know if this is incorrect. I had to make this change because I was getting an error that no proving key existed for the `delegate` action. When I stopped trying to load the nonexistent proving key for that action, it Just Worked™.

Reply:
@TalDerei might you know about this? Are there some actions that do not require proving keys? (aka delegation?)

Reply:
> @TalDerei might you know about this? Are there some actions that do not require proving keys? (aka delegation?)

The 7 action types (excluding `nullifier_derivation` which is used for verifiable disclosure purposes, and won't be used by the web-code) in proving-keys.ts [require](https://github.com/penumbra-zone/penumbra/blob/a9694cfc997a893bf3a9a66a6822e2b078b861b4/crates/bin/pcli/tests/proof.rs#L99) proving keys fetched from [here](https://github.com/penumbra-zone/penumbra/tree/a9694cfc997a893bf3a9a66a6822e2b078b861b4/crates/crypto/proof-params/src/gen). These are the only actions that require proving keys, and there are other [actions](https://github.com/penumbra-zone/penumbra/blob/a9694cfc997a893bf3a9a66a6822e2b078b861b4/crates/core/transaction/src/plan/action.rs#L38) that don't.

Reply:
Yes, most actions don't have proofs, so we should be handling the case that an action doesn't require a proving key.

Wondering if it might be cleanest to have a mapping of action type -> proving key name | null ?